### PR TITLE
Standalone CLI python tests

### PR DESCRIPTION
**Pull request by ni4**
As it was discussed in the previous PR #327 we agreed to have CLI tests as standalone python script(s).

This pull request resolves this by adding two python scripts in /src/tests:
`./cli_tests.py` - script which should be run with `[rnp | rnpkeys | all]` parameter, and perform testing of the corresponding utility. Now the following tests are available:
- RSA key generation, import/export to GPG
- encryption/decryption against GPG
- signing/verification against GPG
All tests use different options (symmetric algos, key bits, compression, etc), however they would be extended soon as well as number of tests.

`./perf_tests.py` - some basic at the moment test which compares performance of RNP against the GPG for the small and large file encryption. It will be extended later since now more focus was on `cli_tests.py`

Both scripts should be run from `/src/tests/`

### CODE REVIEW

**Comment by dewyatt** [comment]
I think I managed to run into that issue you mentioned in #353 while running cli_tests.py.

Anyways, my first impression is that this looks great!

I guess there's some comparisons that maybe aren't valid yet, like rnp doesn't support compression, etc. But it's pretty nice to have the support there for that comparison.

File: src/tests/cli_tests.py [hunk 1]
```diff
@@ -0,0 +1,454 @@
+#!/usr/bin/python
```
Comment (dewyatt) [inline src/tests/cli_tests.py]: On some newer OSes python == python3, so I usually go with `#!/usr/bin/env python2` (or even `#!/usr/bin/python2`).
Is this OK or does it cause any issues on macOS?

Comment (ni4) [inline src/tests/cli_tests.py]: On macOS I have only `/usr/bin/python2.7` and `/usr/local/bin/python2`. So, maybe, when integrating this to Travis we should better run it as 'python2 cli_tests.py all'? Or there are other ideas how to make it to work everywhere.

**Comment by ni4** [comment]
@dewyatt Hm, `rnp` at least supports decompression. Also `rnp` by default compresses data (and we actually don't have a way to enable/disable compression or change the compression level yet):
```
# off=0 ctb=c1 tag=1 hlen=3 plen=268 new-ctb
:pubkey enc packet: version 3, algo 1, keyid D78F04BE3AF65209
	data: [2048 bits]
# off=271 ctb=d2 tag=18 hlen=2 plen=71 new-ctb
:encrypted data packet:
	length: 71
	mdc_method: 2
# off=284 ctb=c8 tag=8 hlen=2 plen=36 new-ctb
:compressed packet: algo=2
# off=287 ctb=cb tag=11 hlen=2 plen=27 new-ctb
:literal data packet:
	mode b (62), created 1502000813, name="hello.txt",
	raw data: 12 bytes
```

File: src/tests/cli_tests.py (hunk 1 above)
Comment (ronaldtse) [inline src/tests/cli_tests.py]: @ni4 `#!/usr/bin/env python2` works with my macOS which has Python 2 installed via `brew`.

**Comment by ronaldtse** [comment]
Thank you @ni4 ! This is what we really need and gives a good framework for adding more tests.

**Comment by dewyatt** [comment]
@ni4 I just meant that we don't have comparable compression support, like selecting which alg, varying compression levels, etc.

**Pull request merged**